### PR TITLE
Changes in Hyperspectral_{metadata, calculation}.py

### PR DESCRIPTION
**Pull request by FlyingWithJerome**
This update includes the following fixes:

**On [hyperspectral_metadata.py](../blob/master/hyperspectral/hyperspectral_calculation.py)**

- Add more keywords to recognize the gantry time in the metadata. Now "timestamp", "time" and "Time" are all keywords. This change is based on the report from @max-zilla in the [computing-pipeline issue #230](https://github.com/terraref/computing-pipeline/issues/230). Max spotted it in the file from 12/05.

- Refactor part of the codes (to remove redundant) 

**On [hyperspectral_calculation.py](../blob/master/hyperspectral/hyperspectral_calculation.py)**

- From the same [issue](https://github.com/terraref/computing-pipeline/issues/230) above I notice that some of the metadata file DOES NOT have gantry position and speed information, which makes the pixels untraceable. In this case, the calculation module will be disabled and returns all zeros for longitude, latitude and bounding box. Google Map locating is also not available.

**Testing**
- The updates had already been tested with VNIR data from 10-06 and 12-05.

**Bottom Line**
Probably we need some standards for the metadata files. Too many customized cases start to weaken the maintainability of the program

### CODE REVIEW

File: hyperspectral/hyperspectral_calculation.py [hunk 1]
```diff
@@ -54,10 +54,14 @@ def pixel2Geographic(jsonFileLocation, headerFileLocation, cameraOption):
         if "position x [m]" in master["gantry_system_variable_metadata"]:
             x_gantry_pos = float(master["gantry_system_variable_metadata"]["position x [m]"])
             y_gantry_pos = float(master["gantry_system_variable_metadata"]["position y [m]"])
-        else:
+
```
Comment (dlebauer) [inline hyperspectral/hyperspectral_calculation.py]: These changes don't seem major, and they generally seem sensible.

File: hyperspectral/hyperspectral_calculation.py [hunk 2]
```diff
@@ -54,10 +54,14 @@ def pixel2Geographic(jsonFileLocation, headerFileLocation, cameraOption):
...
+        elif "Position x [m]" in master["gantry_system_variable_metadata"]:
             x_gantry_pos = float(master["gantry_system_variable_metadata"]["Position x [m]"])
             y_gantry_pos = float(master["gantry_system_variable_metadata"]["Position y [m]"])
 
+        else: # We notice that there are cases that no position data available
+            return [0], [0], ["0, 0"]*4 , ""
```
Comment (dlebauer) [inline hyperspectral/hyperspectral_calculation.py]: would it make sense to have a missing data value here? otherwise, it is possible to confuse 0,0 with an actual position on the field.

Comment (czender) [inline hyperspectral/hyperspectral_calculation.py]: yes, @FlyingWithJerome will add _FillValue to variables that might have them.
That can be done in a different PR.

**Comment by czender** [comment]
Looks good to me

**Comment by dlebauer** [comment]
@czender instead of closing the pull request you need to merge the pull request ... I've done this. If that option wasn't available to you it should be (just a matter of fixing the permissions).

**Comment by czender** [comment]
Ooops, i could swear i merged it. @FlyingWithJerome was sitting here with me. Thanks for catching that.
